js/widgets/listview.js

```diff
@@ -85,123 +88,126 @@ $.widget( "mobile.listview", $.extend( {
 	refresh: function( create ) {
 		var buttonClass, pos, numli, item, itemClass, itemTheme, itemIcon, icon, a,
 			isDivider, startCount, newStartCount, value, last, splittheme, splitThemeClass, spliticon,
-			altButtonClass, dividerTheme, li,
+			altButtonClass, dividerTheme, li, ol, start, itemClassDict, countBubbles, countTheme, countThemeClass,
 			o = this.options,
-			$list = this.element,
-			ol = !!$.nodeName( $list[ 0 ], "ol" ),
-			start = $list.attr( "start" ),
-			itemClassDict = {},
-			countBubbles = $list.find( ".ui-li-count" ),
-			countTheme = getAttr( $list[ 0 ], "counttheme" ) || this.options.countTheme,
+			$list = this.element;
+
+		if ( !o.enhanced ) {
```

Comment:
The ```enhanced``` option cannot play a role in ```refresh()```, because then subsequent calls to ```refresh()``` won't update the list. The correct way to handle the ```enhanced``` option is for ```_create()``` to not call ```refresh()``` if ```enhanced``` is ```true```.